src/libs/actions/Report.ts

```diff
@@ -1607,6 +1609,39 @@ function readNewestAction(reportID: string | undefined, shouldResetUnreadMarker
     }
 }
 
+function markAllMessagesAsRead() {
+    const lastReadTime = DateUtils.getDBTimeWithSkew();
+
+    const optimisticData: OnyxUpdate[] = [];
+    const reportIDs: string[] = [];
+    Object.keys(allReports ?? {}).forEach((key: string) => {
+        const report = allReports?.[key];
+        const oneTransactionThreadReportID = ReportActionsUtils.getOneTransactionThreadReportID(
+            report?.reportID,
+            allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report?.reportID}`],
+        );
+        const oneTransactionThreadReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${oneTransactionThreadReportID}`];
+        if (!report || !isUnread(report, oneTransactionThreadReport)) {
+            return;
+        }
+
+        optimisticData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`,
+            value: {
+                lastReadTime,
+            },
+        });
+        reportIDs.push(report.reportID);
+    });
+
+    const parameters: MarkAllMessagesAsReadParams = {
+        reportIDs,
+    };
+
+    API.write(WRITE_COMMANDS.MARK_ALL_MESSAGES_AS_READ, parameters, {optimisticData});
+}
```

Comment:
Before calling the API, let's make sure the `reportIDs` array is not empty

Reply:
Done